lib/puppet/application/agent.rb

```diff
@@ -71,6 +71,10 @@ def preinit
     options[:waitforcert] = arg.to_i
   end
 
+  option("--job-id ID") do |arg|
+    options[:job_id] = arg
```

Comment:
This data isn't intended to be particularly trusted. Agents could spoof having been part of a job, but only in PuppetDB. Orchestrator is the official record for which jobs happened, and there's no way to forge that. Threading the data through PuppetDB is a convenience to expose it via PQL, etc.

The usage pattern mostly involves querying PuppetDB to find the job you care about (I'm looking at a report and want to know if it was part of a job) and then querying orchestrator to find out what happened in that job. If a node were to lie about being part of the job, you would learn that when asking orchestrator.